ci-operator/step-registry/cucushift/hypershift-extended/enable-dns/custom-kubeconfig/cucushift-hypershift-extended-enable-dns-custom-kubeconfig-commands.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ ! -f "${SHARED_DIR}/kubeconfig" ]; then
+    exit 1
+fi
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
```

Comment:
```suggestion
KUBECONFIG="${SHARED_DIR}/kubeconfig"
CUSTOM_KUBECONFIG="/tmp/custom_kube"
SECRET_NAME="${CLUSTER_NAME}-custom-admin-kubeconfig"
HCP_NAMESPACE="clusters-${CLUSTER_NAME}"

if [ ! -f "$KUBECONFIG" ]; then
    echo "ERROR: Kubeconfig not found" >&2
    exit 1
fi

export KUBECONFIG

[ -f "${SHARED_DIR}/proxy-conf.sh" ] && source "${SHARED_DIR}/proxy-conf.sh"
```

Reply:
I think steps `custom-kubeconfig`, `generate-certificate` and `update-record`,  If they are not reusable and closely related to each other, it can be considered to merge them into one: `enable-dns`